### PR TITLE
LPS-164434 | Add tests CanRetainNonDeletedFieldValue, CSSURLCanBeRepeated, JSURLCanBeRepeatedMultipleTimes and macro removeCSSURLRow

### PR DESCRIPTION
**Pull request by bronsonsoda**
**Background**
Create remote test that ensures CSSURL can be repeated

**Test plan**
_CSSURLCanBeRepeated_
Given Remote Application > Custom Elements create page
When click repeatable button for CSS URL
Then new JS URL field is added
And Then repeatable button + and - are present for both CSS URL fields
https://issues.liferay.com/browse/LPS-164434

**Background**
Create remote test that ensures JSURL can be repeated

**Test plan**
_JSURLCanBeRepeatedMultipleTimes_
Given Remote Application > Custom Elements create page
When click repeatable button 3 times for JS URL
Then 3 new JS URL field are added
https://issues.liferay.com/browse/LPS-164435

**Background**
Create remote test that ensures non deleted field value is retained

**Test plan**
_CanRetainNonDeletedFieldValue_
Given Custom Element with two valid JS URLs
When delete first repeatable field
Then only 1 JS URL field is present
And Then the JS URL value matches the 2nd JS URL value (prior to deletion)
And Then repeatable button - (minus) is not present
https://issues.liferay.com/browse/LPS-164436

### CODE REVIEW

**Comment by bronsonsoda** [comment]
"And Then the JS URL value matches the 2nd JS URL value (prior to deletion)"

I'm not 100% sure on this part

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by john-co** [comment]
ci:test:asdf

**Comment by john-co** [comment]
Just started reviewing :)

**Review by john-co — CHANGES_REQUESTED** [review]
Please see comments

**Comment by bronsonsoda** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by bronsonsoda** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by bronsonsoda** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 1]
```diff
@@ -36,6 +36,95 @@ definition {
...
+				key_index = "2",
+				locator1 = "RemoteAppsEntry#URL_ROW_INDEX");
+		}
+
+		task ("And Then the JS URL value matches the 2nd JS URL value (prior to deletion)") {
+			AssertElementPresent(
+				key_id = "ClientExtensionAdminPortlet_url",
+				key_index = "2",
+				locator1 = "RemoteAppsEntry#URL_ROW_INDEX");
+		}
```
[the remark this answers is not in the document]
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:86]: double check this

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 2]
```diff
@@ -36,6 +36,108 @@ definition {
...
+
+			Type(
+				key_id = "ClientExtensionAdminPortlet_url",
+				key_index = "2",
+				locator1 = "RemoteAppsEntry#URL_ROW_INDEX",
+				value1 = "https://www.liferay.com/company/our-story");
+		}
+
+		task ("When delete first repeatable field") {
+			RemoteAppsEntry.removeURLRow(key_index = "2");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:65]: key_index = "1"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase (hunk 1 above)
[the remark this answers is not in the document]
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:86]: assert `https://www.liferay.com/company/our-story` is present

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 3]
```diff
@@ -36,6 +36,109 @@ definition {
...
+				key_index = "1",
+				key_text = "https://www.liferay.com/",
+				locator1 = "RemoteAppsEntry#URL_ROW_INDEX");
+		}
+
+		task ("And Then the JS URL value matches the 2nd JS URL value (prior to deletion)") {
+			AssertElementPresent(
+				key_id = "ClientExtensionAdminPortlet_url",
+				key_index = "2",
+				key_text = "https://www.liferay.com/",
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:85]: assert this one `https://www.liferay.com/company/our-story`

**Comment by bronsonsoda** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by bronsonsoda** [comment]
I tried **AssertNotVisible** and **IsNotVisible** but only **IsNotVisible** work 🤔

**Comment by bronsonsoda** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by john-co** [comment]
ci:test:frontend-remote-apps:nocompile

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**